lib/matplotlib/collections.pyi

```diff
@@ -51,10 +51,10 @@ class Collection(artist.Artist, cm.ScalarMappable):
     def get_offsets(self) -> ArrayLike: ...
     def set_linewidth(self, lw: float | Sequence[float]) -> None: ...
     def set_linestyle(self, ls: LineStyleType | Sequence[LineStyleType]) -> None: ...
-    def set_capstyle(self, cs: CapStyle | Sequence[CapStyle]) -> None: ...
-    def get_capstyle(self) -> CapStyle | Sequence[CapStyle]: ...
-    def set_joinstyle(self, js: JoinStyle | Sequence[JoinStyle]) -> None: ...
-    def get_joinstyle(self) -> JoinStyle | Sequence[JoinStyle]: ...
+    def set_capstyle(self, cs: CapStyleType) -> None: ...
+    def get_capstyle(self) -> Literal["butt", "projecting", "round"]: ...
+    def set_joinstyle(self, js: JoinStyleType) -> None: ...
+    def get_joinstyle(self) -> Literal["miter", "round", "bevel"]: ...
```

Comment:
No, actually, that is intentional, that should never have included Sequence, as that is not actually supported:

From the Collection docstring:
```
    Most properties of a collection can be configured per-element. Therefore,
    Collections have "plural" versions of many of the properties of a `.Patch`
    (e.g. `.Collection.get_paths` instead of `.Patch.get_path`). Exceptions are
    the *zorder*, *hatch*, *pickradius*, *capstyle* and *joinstyle* properties,
    which can only be set globally for the whole collection.
```

When looking at the implementation again, it was also clear that that was errant.